collector/powersupplyclass.go

```diff
@@ -0,0 +1,194 @@
+			"time_to_empty_seconds": powerSupply.TimeToEmptyNow,
+			"time_to_full_seconds":  powerSupply.TimeToFullNow,
+		} {
+			if value != nil {
+				pushPowerSupplyMetric(ch, c.subsystem, name, float64(*value), powerSupply.Name, prometheus.GaugeValue)
+			}
+		}
+
+		for name, value := range map[string]*int64{
+			"current_boot":                powerSupply.CurrentBoot,
+			"current_max":                 powerSupply.CurrentMax,
+			"current_ampere":              powerSupply.CurrentNow,
+			"energy_empty":                powerSupply.EnergyEmpty,
+			"energy_empty_design":         powerSupply.EnergyEmptyDesign,
+			"energy_full":                 powerSupply.EnergyFull,
+			"energy_full_design":          powerSupply.EnergyFullDesign,
+			"energy_watthour":             powerSupply.EnergyNow,
+			"voltage_boot":                powerSupply.VoltageBoot,
+			"voltage_max":                 powerSupply.VoltageMax,
+			"voltage_max_design":          powerSupply.VoltageMaxDesign,
+			"voltage_min":                 powerSupply.VoltageMin,
+			"voltage_min_design":          powerSupply.VoltageMinDesign,
+			"voltage_volt":                powerSupply.VoltageNow,
+			"voltage_ocv":                 powerSupply.VoltageOCV,
+			"charge_control_limit":        powerSupply.ChargeControlLimit,
+			"charge_control_limit_max":    powerSupply.ChargeControlLimitMax,
+			"charge_counter":              powerSupply.ChargeCounter,
+			"charge_empty":                powerSupply.ChargeEmpty,
+			"charge_empty_design":         powerSupply.ChargeEmptyDesign,
+			"charge_full":                 powerSupply.ChargeFull,
+			"charge_full_design":          powerSupply.ChargeFullDesign,
+			"charge_ampere":               powerSupply.ChargeNow,
+			"charge_term_current":         powerSupply.ChargeTermCurrent,
+			"constant_charge_current":     powerSupply.ConstantChargeCurrent,
+			"constant_charge_current_max": powerSupply.ConstantChargeCurrentMax,
+			"constant_charge_voltage":     powerSupply.ConstantChargeVoltage,
+			"constant_charge_voltage_max": powerSupply.ConstantChargeVoltageMax,
+			"precharge_current":           powerSupply.PrechargeCurrent,
+			"input_current_limit":         powerSupply.InputCurrentLimit,
+			"power_watt":                  powerSupply.PowerNow,
+		} {
+			if value != nil {
+				pushPowerSupplyMetric(ch, c.subsystem, name, float64(*value)/1e6, powerSupply.Name, prometheus.GaugeValue)
+			}
+		}
+
+		for name, value := range map[string]*int64{
+			"temp_celsius":     powerSupply.Temp,
+			"temp_alert_max":   powerSupply.TempAlertMax,
+			"temp_alert_min":   powerSupply.TempAlertMin,
+			"temp_ambient":     powerSupply.TempAmbient,
+			"temp_ambient_max": powerSupply.TempAmbientMax,
+			"temp_ambient_min": powerSupply.TempAmbientMin,
+			"temp_max":         powerSupply.TempMax,
+			"temp_min":         powerSupply.TempMin,
+		} {
+			if value != nil {
+				pushPowerSupplyMetric(ch, c.subsystem, name, float64(*value)*10, powerSupply.Name, prometheus.GaugeValue)
```

Comment:
Shouldn't this be `float64(*value)/10.0` to convert tenth-degrees to degrees?

Reply:
@SuperQ Yeah, you are right. I changed it.